test/d2l-filter-dropdown/d2l-filter-dropdown.js

```diff
@@ -1,387 +1,155 @@
-			_getExpectedAndImport(filter);
-			window.requestAnimationFrame(function() {
-				var totalCount = _getDropdownOpener();
-				assert.equal('Filter', totalCount.text);
+		test('closing the filter triggers the d2l-filter-dropdown-close event', function(done) {
+			var dropdown = filter.shadowRoot.querySelector('d2l-dropdown-tabs');
+			filter.addEventListener('d2l-filter-dropdown-close', function() {
 				done();
 			});
-		});
-		test('total filter count is displayed correctly when one option is selected', function(done) {
-			_setSelectedOptions([2]);
-			_getExpectedAndImport(filter);
-			window.requestAnimationFrame(function() {
-				var totalCount = _getDropdownOpener();
-				assert.equal('Filter: 1 Filter', totalCount.text);
-				done();
+
+			dropdown.toggleOpen();
+
+			requestAnimationFrame(function() {
+				requestAnimationFrame(function() {
+					dropdown.toggleOpen();
+				});
 			});
 		});
-		test('total filter count is displayed correctly when multiple options are selected', function(done) {
-			var selected = [0, 4, 5];
-			_setSelectedOptions(selected);
-			_getExpectedAndImport(filter);
-			window.requestAnimationFrame(function() {
-				var totalCount = _getDropdownOpener();
-				assert.equal(`Filter: ${selected.length} Filters`, totalCount.text);
-				done();
-			});
+		test('the opener text is set accordingly', function() {
+			var openerButton = filter.shadowRoot.querySelector('d2l-dropdown-button-subtle');
+			assert.equal(openerButton.text, 'Filter: 3 Filters');
+
+			filter.totalSelectedOptionCount = 1;
+			assert.equal(openerButton.text, 'Filter: 1 Filter');
+
+			filter.totalSelectedOptionCount = 0;
+			assert.equal(openerButton.text, 'Filter');
 		});
 		test('filter categories display correctly', function(done) {
-			_getExpectedAndImport(filter);
-			window.requestAnimationFrame(function() {
-				var headers = _getTabPanels();
-				assert.equal(filter._filters.length, headers.length);
-				for (var i = 0; i < headers.length; i++) {
-					assert.equal(filter._filters[i].title, headers[i].text);
+			function getTabs() {
+				var tabs = filter.shadowRoot.querySelector('d2l-tabs').shadowRoot.querySelectorAll('d2l-tab');
+				if (tabs.length === 2) {
+					assert.equal(tabs[0].text, 'Category 1 (3)');
+					assert.equal(tabs[1].text, 'Category 2');
+					done();
+				} else {
+					setTimeout(getTabs, 25);
 				}
-				done();
-			});
+			}
+
+			getTabs();
 		});
-		test('filter options display correctly', function(done) {
-			_getExpectedAndImport(filter);
-			window.requestAnimationFrame(function() {
-				var pages = _getPages();
-				for (var i = 0; i < pages.length; i++) {
-					var opts = _getOptions(pages[i]);
-					assert.equal(pages[i].options.length, opts.length);
-					for (var j = 0; j < opts.length; j++) {
-						assert.equal(pages[i].options[j].title, opts[j].text);
-					}
-				}
-				done();
-			});
+	});
+
+	suite('d2l-filter-dropdown-category', function() {
```

Comment:
nit: this deserves its own file for searchability (looking at the file system you'd assume dropdown-category has no tests)